src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1503,13 +1610,31 @@ public static void mutateAtomically(List<Mutation> mutations,
                     if (accordResult != null)
                     {
                         IAccordService accord = AccordService.instance();
-                        TxnResult.Kind kind = accord.getTxnResult(accordResult, true, consistencyLevel, requestTime).kind();
+                        TxnResult.Kind kind = accord.getTxnResult(accordResult).kind();
                         if (kind == retry_new_protocol && failure == null)
                             continue;
                         Tracing.trace("Successfully wrote Accord mutations");
                         cleanup.ackMutation();
                     }
                 }
+                catch (TopologyMismatch e)
+                {
+                    // Don't suppress existing failure
+                    if (failure == null)
+                    {
+                        // For now assuming topology mismatch is caused by a race misrouting
+                        Tracing.trace("Accord returned topology mismatch, retrying: " + e.getMessage());
+                        logger.debug("Accord returned topology mismatch, retrying: " + e.getMessage());
+                        continue;
```

Comment:
stated above.  This event isn't a safe one to retry on, this can lead to an infinite loop